### PR TITLE
Add section about inference endpoints in HF guide

### PR DESCRIPTION
**Pull request by osanseviero**

### CODE REVIEW

File: docs/pages/docs/guides/huggingface.mdx [hunk 1]
```diff
@@ -232,6 +236,23 @@ export default function Completion() {
 
 </Steps>
 
+## Guide: Using Production-ready Inference Endpoints
+
+Inference Endpoints offers a secure solution to deploy models from the Hub on dedicated and autoscaling infrastructured. It supports popular libraries such as `transformers` and `diffusers` out-of-the-box, but users can also implement custom handlers for their own solutions. Read more about Inference Endpoints in [their documentation](https://huggingface.co/docs/inference-endpoints/index).
+ With Inference Endpoints, you can even use state of the art open-source LLMs requiring more compute such as [Falcon 40B](https://huggingface.co/blog/inference-endpoints-llm).
```
Comment (xenova) [inline docs/pages/docs/guides/huggingface.mdx:242]:
```suggestion
 With Inference Endpoints, you can even use state-of-the-art open-source LLMs requiring more compute such as [Falcon 40B](https://huggingface.co/blog/inference-endpoints-llm).
```

Lots of hypens, but I think it's supposed to be hyphenated when using as a compound adjective.

File: docs/pages/docs/guides/huggingface.mdx [hunk 2]
```diff
@@ -69,7 +72,8 @@ export async function POST(req: Request) {
   // Extract the `messages` from the body of the request
   const { messages } = await req.json()
 
-  const response = Hf.textGenerationStream({
+  // Initialize a text generation stream using Hugging Face Inference SDK
```
Comment (xenova) [inline docs/pages/docs/guides/huggingface.mdx:75]:
```suggestion
  // Initialize a text-generation stream using the Hugging Face Inference SDK
```

File: docs/pages/docs/guides/huggingface.mdx [hunk 3]
```diff
@@ -232,6 +236,23 @@ export default function Completion() {
 
 </Steps>
 
+## Guide: Using Production-ready Inference Endpoints
+
+Inference Endpoints offers a secure solution to deploy models from the Hub on dedicated and autoscaling infrastructured. It supports popular libraries such as `transformers` and `diffusers` out-of-the-box, but users can also implement custom handlers for their own solutions. Read more about Inference Endpoints in [their documentation](https://huggingface.co/docs/inference-endpoints/index).
```
Comment (xenova) [inline docs/pages/docs/guides/huggingface.mdx:241]:
```suggestion
Inference Endpoints offers a secure solution to deploy models from the Hub on dedicated and autoscaling infrastructure. It supports popular libraries such as `transformers` and `diffusers` out-of-the-box, but users can also implement custom handlers for their own solutions. Read more about Inference Endpoints in [their documentation](https://huggingface.co/docs/inference-endpoints/index).
```

**Review by xenova — COMMENTED** [review]
Looks good! 🚀  (just a few minor spelling/grammar suggestions)

**Review by MaxLeiter — APPROVED** [review]

**Comment by MaxLeiter** [comment]
Thank you!

**Pull request merged**